### ISSUE TITLE
custom headers

### ISSUE THREAD

**Issue by vasilakisfil** [wafflespeanut/paperclip#208]
From what I understand, this library doesn't support Authorization yet (like basic/token/bearer) through the security and securityDefinitions of swagger. Is there a way to inject custom headers when trying to send a request ?

I would suggest so support that feature, even if you implement the seuciry/securityDefinitions of swagger, it can be handy at some cases.

**Comment by wafflespeanut** [comment]
You're right that paperclip doesn't support security/securityDefinitions yet (it'll be part of the next release). Per-operation headers are not possible as of now (I'll certainly take a look), but there's a workaround for global headers right now.

Anything that implements [`ApiClient` trait](https://github.com/wafflespeanut/paperclip/blob/0dc323f8004f235e44d2eaf0c4fe28f0bf9819eb/src/build/client_mod.hbs#L176) is accepted as a client. This can be found in the generated code. It contains the request builder method, which can be used to inject global headers. For example,

```rust
struct MyClient(reqwest::Client);

#[async_trait::async_trait]
impl ApiClient for MyClient {
    type Request = reqwest::RequestBuilder;
    type Response = reqwest::Response;

    fn request_builder(&self, method: http::Method, rel_path: &str) -> Self::Request {
        let req = self.0.request_builder(method, rel_path);
        // add global headers?
        req
    }

    async fn make_request(&self, req: Self::Request) -> Result<Self::Response, ApiError<Self::Response>> {
        Ok(self.0.make_request(req).await?)
    }
}
```

This client can then be passed to `send` method.
